plugins/pulp_rpm/plugins/importers/yum/existing.py

```diff
@@ -112,24 +112,13 @@ def check_all_and_associate(wanted, sync_conduit):
         model = plugin_api.get_unit_model_by_id(unit_type)
         # FIXME "fields" does not get used, but it should
         # fields = model.unit_key_fields + ('_storage_path',)
-        rpm_srpm_drpm = unit_type in (ids.TYPE_ID_RPM,
-                                      ids.TYPE_ID_SRPM,
-                                      ids.TYPE_ID_DRPM)
-
-        unit_generator = (model(**unit_tuple._asdict()) for unit_tuple in values)
+        unit_generator = (model(**unit_tuple._asdict()) for unit_tuple in values.copy())
         for unit in units_controller.find_units(unit_generator):
-            if rpm_srpm_drpm:
```

Comment:
I think this is still important if the download policy is not deferred. To preserve the current sync behavior, I think we should keep this logic if downloading is not deferred.